flake.nix

```diff
@@ -1,31 +1,95 @@
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    advisory-db = {
+      url = "github:rustsec/advisory-db";
+      flake = false;
+    };
   };
 
-  outputs = { self, flake-utils, naersk, nixpkgs }:
+  outputs = { self, flake-utils, rust-overlay, crane, advisory-db, nixpkgs, }:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = (import nixpkgs) {
           inherit system;
+          overlays = [ (import rust-overlay) ];
         };
-        naersk' = pkgs.callPackage naersk {};
-        nativeBuildInputs = with pkgs; [ pkg-config ];
-      in rec {
-        packages.default = naersk'.buildPackage {
-          inherit nativeBuildInputs;
-          src = ./.;
-          buildInputs = with pkgs; [ openssl ] ++ (pkgs.lib.lists.optionals (stdenv.isDarwin) [ darwin.apple_sdk.frameworks.SystemConfiguration ]);
+        inherit (pkgs) lib callPackage;
+        rustToolchain = callPackage ./nix/toolchain.nix { };
+
+        darwinFrameworks = with pkgs.darwin.apple_sdk.frameworks; [
+          Security
+          SystemConfiguration
+        ];
+
+        devTools = [ rustToolchain ];
+        libgit2_1_7_2 = callPackage ./nix/libgit2.nix {
+          inherit (pkgs.darwin.apple_sdk.frameworks) Security;
         };
-        devShells.default = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; nativeBuildInputs ++ [
-            cargo
-            rustc
-          ];
+
+        dependencies = with pkgs;
+          [ libgit2_1_7_2 libiconv openssl openssl.dev ]
+          ++ lib.lists.optionals stdenv.isDarwin darwinFrameworks;
+
+        nativeBuildInputs = with pkgs; [ pkg-config ] ++ dependencies;
+
+        buildEnvVars = {
+          NIX_LDFLAGS = [ "-L" "${pkgs.libiconv}/lib" ];
+
+          LIBGIT2_NO_VENDOR = 1;
+          OPENSSL_NO_VENDOR = 1;
+        };
+
+        buffrs = callPackage ./nix/buffrs.nix {
+          inherit crane advisory-db buildEnvVars nativeBuildInputs
+            rustToolchain;
+          buildInputs = [ rustToolchain ];
         };
-        checks.builds = packages.default;
-      }
-    );
+      in {
+        # NB: if this does not build and you need to modify the file,
+        #     please ensure you also make the corresponding changes in the devshell
+        packages.default = buffrs.package;
+        apps.default = flake-utils.lib.mkApp { drv = buffrs.package; };
+
+        devShells.default = pkgs.mkShell ({
+          inherit nativeBuildInputs;
+          buildInputs = devTools ++ dependencies;
+        } // buildEnvVars);
+
+        formatter = with pkgs;
+          writeShellApplication {
+            name = "nixfmt-nix-files";
+            runtimeInputs = [ fd nixfmt ];
+            text = "fd \\.nix\\$ --hidden --type f | xargs nixfmt";
+          };
+
+        checks = ({
```

Comment:
i think you can just run `nixfmt -c .` for this whole thing?

Reply:
It will pick up non-nix files then

Reply:
The mac run now takes 9 mins if you need to rebuild the deps.

Building the deps took at least 5 mins. If you don't need to change them, you'll use the cache and ideally get an appropriate speedup.